kedro-datasets/setup.py

```diff
@@ -26,6 +26,14 @@ def _collect_requirements(requires):
 }
 holoviews_require = {"holoviews.HoloviewsWriter": ["holoviews~=1.13.0"]}
 matplotlib_require = {"matplotlib.MatplotlibWriter": ["matplotlib>=3.0.3, <4.0"]}
+netcdf_require = {
+    "netcdf.NetCDFDataset": [
+        "h5netcdf>=1.2.0",
```

Comment:
Notice that this does not support Python 3.8:

```
ERROR: Ignored the following versions that require a different python version: 1.2.0 Requires-Python >=3.9
ERROR: Could not find a version that satisfies the requirement h5netcdf>=1.2.0; extra == "netcdf" (from kedro-datasets[netcdf]) (from versions: 0.1, 0.1.1, 0.1.2, 0.1.3, 0.2.0, 0.2.1, 0.2.2, 0.3.0, 0.3.1, 0.4.0, 0.4.1, 0.4.2, 0.4.3, 0.5.0, 0.5.1, 0.6.0, 0.6.1, 0.6.2, 0.7.0, 0.7.1, 0.7.3, 0.7.4, 0.8.0, 0.8.1, 0.9.0, 0.10.0, 0.11.0, 0.12.0, 0.13.0, 0.13.1, 0.14.0, 0.14.1, 0.15.0, 1.0.0, 1.0.1, 1.0.2, 1.1.0)
ERROR: No matching distribution found for h5netcdf>=1.2.0; extra == "netcdf"
```

Reply:
Good eye. Will fix this similar to how I did the xarray version locking.